### ISSUE TITLE
Eredar Bloodmage 117782

### ISSUE THREAD

**Issue by Ulduar** [AshamaneProject/AshamaneCore#680]
**Description:**
Simple thrash npc have one spell cast, but spell need script.

```SQL
-- Eredar Bloodmage SAI
SET @ENTRY := 117782;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@ENTRY;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,0,0,0,0,0,100,0,3000,4500,21000,23000,11,235222,0,0,0,0,0,5,0,0,0,0,0,0,0,"Eredar Bloodmage -  IC Timer - Cast Blood Siphon.");
```
Caster cast on target. Target get damage from CurrentHealthPct. After damage target cast heal spell (heal also from CurHealthPct) on enemies.
**Usefull Wowhead links:**
http://www.wowhead.com/npc=117782/eredar-bloodmage
http://www.wowhead.com/spell=235222/blood-siphon
http://www.wowhead.com/spell=235232/blood-siphon
http://www.wowhead.com/spell=235262/blood-siphon

My variant (need check)
``` CPP
enum EredarBloodmage
{
    SPELL_BLOOD_SIPHON_DAMAGE  = 235232,
    SPELL_BLOOD_SIPHON_HEAL    = 235262
};

// 235222
class spell_gen_eredar_bloodmage_blood_siphon : public SpellScriptLoader
{
    public:
        spell_gen_eredar_bloodmage_blood_siphon() : SpellScriptLoader("spell_gen_eredar_bloodmage_blood_siphon") { }

        class spell_gen_eredar_bloodmage_blood_siphon_AuraScript : public AuraScript
        {
            PrepareAuraScript(spell_gen_eredar_bloodmage_blood_siphon_AuraScript);

            void HandleEffectPeriodic(AuraEffect const* /*aurEff*/)
            {
                PreventDefaultAction();
                GetCaster()->CastSpell((GetHitUnit(), aurEff->GetAmount(), true);
            }

            void Register() override
            {
                OnEffectPeriodic += AuraEffectPeriodicFn(spell_gen_eredar_bloodmage_blood_siphon_AuraScript::HandleEffectPeriodic, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY);
            }
        };

        AuraScript* GetAuraScript() const override
        {
            return new spell_gen_eredar_bloodmage_blood_siphon_AuraScript();
        }
};

// 235232 - Blood Siphon (damage)
class spell_gen_eredar_bloodmage_blood_siphon_damage : public SpellScriptLoader
{
public:
    spell_gen_eredar_bloodmage_blood_siphon_damage() : SpellScriptLoader("spell_gen_eredar_bloodmage_blood_siphon_damage") { }

    class spell_gen_eredar_bloodmage_blood_siphon_damage_SpellScript : public SpellScript
    {
        PrepareSpellScript(spell_gen_eredar_bloodmage_blood_siphon_damage_SpellScript);

        void CalculateDamage(SpellEffIndex /*effIndex*/)
        {
            Unit* target = GetHitUnit();
            if (!target)
                return;

	    int32 healthPctDmg = target->CountPctFromCurHealth(GetSpellInfo()->GetEffect(EFFECT_0)->BasePoints);
            SetHitDamage(healthPctDmg);
	    target->CastCustomSpell(GetCaster(), SPELL_BLOOD_SIPHON_HEAL, &healthPctDmg, 0, 0, true);
            
        }

        void Register() override
        {
            OnEffectHitTarget += SpellEffectFn(spell_gen_eredar_bloodmage_blood_siphon_damage_SpellScript::CalculateDamage, EFFECT_0, SPELL_EFFECT_SCHOOL_DAMAGE);
        }
    };

    SpellScript* GetSpellScript() const override
    {
        return new spell_gen_eredar_bloodmage_blood_siphon_damage_SpellScript();
    }
};
```

**Comment by HannibalRoG** [comment]
@Ulduar for c++ patch you need to open a PR instead.

**Comment by Traesh** [comment]
Closed by cc18896313914e555229f767a223822a9c2af2bd
